src/display-api/display-api.service.ts

```diff
@@ -126,4 +133,155 @@ export class DisplayApiService {
       );
     }
   }
+
+  /**
+   * Fetches all screens.
+   */
+  async fetchScreens(): Promise<ScreenScreenJsonld[]> {
+    try {
+      const config = await this.getAuthenticatedConfig();
+      const screensApi = new ScreensApi(config);
+      const screens: ScreenScreenJsonld[] = [];
+
+      let page = 1;
+      const itemsPerPage = "24";
+      let fetch = true;
+      while (fetch) {
+        const response = await screensApi.getV1Screens(page, itemsPerPage);
+        // TODO, figure out how to avoid any.
+        const data = response.data as any;
+        data["hydra:member"].forEach((screen: ScreenScreenJsonld) => {
+          screens.push(screen);
+        });
+        if (data["hydra:member"].length < itemsPerPage) {
+          fetch = false;
+        } else {
+          page += 1;
+        }
+      }
+
+      return screens;
+    } catch (error) {
+      this.logger.error("Error fetching screens", (error as Error).message);
+    }
+
+    return [];
+  }
+
+  /**
+   * Gets a playlist by name.
+   *
+   * @param {string} name the name of the playlist to get
+   */
+  async getPlaylistByName(name: string): Promise<PlaylistPlaylistJsonld | null> {
+    try {
+      const config = await this.getAuthenticatedConfig();
+      const playlistsApi = new PlaylistsApi(config);
+
+      const response = await playlistsApi.getV1Playlists(1, 1, name);
+      const data = response.data as any;
+      const playlist = data["hydra:member"].shift();
+      if (playlist && playlist["@id"]) {
+        return playlist;
+      }
+    } catch (error) {
+      this.logger.error(`Error getting playlist with name "${name}"`, (error as Error).message);
+    }
+
+    return null;
+  }
+
+  /**
+   * Gets the slides in a given playlist.
+   *
+   * @param {string} playlistId the ID of the playlist to get slides for
+   */
+  async getPlaylistSlides(playlistId: string): Promise<PlaylistSlide[]> {
+    const config = await this.getAuthenticatedConfig();
+    const playlistsApi = new PlaylistsApi(config);
+    const slides: PlaylistSlide[] = [];
+
+    let page = 1;
+    const itemsPerPage = "24";
```

Comment:
You have a repeating pattern here - consider some kind of utility class/function that can handle the fetch of paged resources.

Reply:
Ok, maybe we can do that as a followup?